ComputeAdmin/AzureStack.ComputeAdmin.psm1

```diff
@@ -660,5 +662,185 @@ Function CreateGalleyItem{
+
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidateNotNullorEmpty()]
+        [String] $extensionBlobURI,
+
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidateSet('Windows' ,'Linux')]
+        [String] $osType,
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [String] $vmScaleSetEnabled = "false",
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [String] $supportMultipleExtensions = "false",
+    
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [ValidateNotNullorEmpty()]
+        [String] $tenantID,
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [String] $location = 'local',
+
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(Mandatory=$true, ParameterSetName='VMExtesionFromAzure')]
+        [System.Management.Automation.PSCredential] $azureStackCredentials,
+
+        [Parameter(ParameterSetName='VMExtensionFromLocal')]
+        [Parameter(ParameterSetName='VMExtesionFromAzure')]
+        [string] $ArmEndpoint = 'https://adminmanagement.local.azurestack.external'
+    )
+
+    $resourceGroupName = "addvmextresourcegroup"
+    $storageAccountName = "addvmextstorageaccount"
+    $containerName = "addvmextensioncontainer"
+
+    $subscription, $headers =  (Get-AzureStackAdminSubTokenHeader -TenantId $tenantId -AzureStackCredentials $azureStackCredentials -ArmEndpoint $ArmEndpoint)
+
+    #potentially the RG was not cleaned up when exception happened in previous run. Test for exist
+    if (-not (Get-AzureRmResourceGroup -Name $resourceGroupName -Location $location -ErrorAction SilentlyContinue)) {
+        New-AzureRmResourceGroup -Name $resourceGroupName -Location $location 
     }
 
+    #same for storage
+    $storageAccount = Get-AzureRmStorageAccount -Name $storageAccountName -ResourceGroupName $resourceGroupName -ErrorAction SilentlyContinue
+    if (-not ($storageAccount)) {
+        $storageAccount = New-AzureRmStorageAccount -Name $storageAccountName -Location $location -ResourceGroupName $resourceGroupName -Type Standard_LRS
+    }
+
+    Set-AzureRmCurrentStorageAccount -StorageAccountName $storageAccountName -ResourceGroupName $resourceGroupName
+    
+    #same for container
+    $container = Get-AzureStorageContainer -Name $containerName -ErrorAction SilentlyContinue
+    if (-not ($container)) {
+        $container = New-AzureStorageContainer -Name $containerName -Permission Blob
+    }
+
+    if($pscmdlet.ParameterSetName -eq "VMExtensionFromLocal")
```

Comment:
VMExtensionFromLocal seems can be made as a const as it appears many times in comparison expressions